### PR TITLE
feat: Support sparse float vector bulk insert for binlog/json/parquet

### PR DESCRIPTION
**Pull request by cydrain**
Issue: #22837

### CODE REVIEW

**Comment by zhengbuqian** [comment]
issue: #29419

File: internal/util/importutilv2/json/row_parser.go [hunk 1]
```diff
@@ -49,9 +49,12 @@ func NewRowParser(schema *schemapb.CollectionSchema) (RowParser, error) {
 	if err != nil {
 		return nil, err
 	}
-	dim, err := typeutil.GetDim(vecField)
-	if err != nil {
-		return nil, err
+	dim := int64(0)
+	if typeutil.IsVectorType(vecField.DataType) && !typeutil.IsSparseFloatVectorType(vecField.DataType) {
```
Comment (bigsheeper) [inline internal/util/importutilv2/json/row_parser.go:53]: `typeutil.GetDim` already checks "IsVectorType", so how about wrapping `IsSparseFloatVectorType` into `IsVectorType`?

File: internal/util/importutilv2/numpy/field_reader.go [hunk 2]
```diff
@@ -51,7 +51,8 @@ func NewFieldReader(reader io.Reader, field *schemapb.FieldSchema) (*FieldReader
 	}
 
 	var dim int64 = 1
-	if typeutil.IsVectorType(field.GetDataType()) {
+	dataType := field.GetDataType()
+	if typeutil.IsVectorType(dataType) && !typeutil.IsSparseFloatVectorType(dataType) {
```
Comment (bigsheeper) [inline internal/util/importutilv2/numpy/field_reader.go:55]: ditto

File: internal/util/importutilv2/parquet/field_reader.go [hunk 3]
```diff
@@ -47,7 +48,7 @@ func NewFieldReader(ctx context.Context, reader *pqarrow.FileReader, columnIndex
 	}
 
 	var dim int64 = 1
-	if typeutil.IsVectorType(field.GetDataType()) {
+	if typeutil.IsVectorType(field.GetDataType()) && !typeutil.IsSparseFloatVectorType(field.GetDataType()) {
```
Comment (bigsheeper) [inline internal/util/importutilv2/parquet/field_reader.go:51]: ditto

File: internal/util/importutilv2/parquet/field_reader.go [hunk 4]
```diff
@@ -425,25 +423,78 @@ func ReadBinaryData(pcr *FieldReader, dataType schemapb.DataType, count int64) (
...
+		if !isVectorAligned(offsets, pcr.dim, schemapb.DataType_SparseFloatVector) {
+			return nil, merr.WrapErrImportFailed("%s not aligned", schemapb.DataType_SparseFloatVector.String())
+		}
+		uint8Reader, ok := listReader.ListValues().(*array.Uint8)
+		if !ok {
+			return nil, WrapTypeErr("binary", listReader.ListValues().DataType().Name(), pcr.field)
+		}
+		vecData := uint8Reader.Uint8Values()
+		for i := 0; i < rows; i++ {
+			elemCount := int((offsets[i+1] - offsets[i]) / 8)
```
Comment (bigsheeper) [inline internal/util/importutilv2/parquet/field_reader.go:446]: `i+1` may be out of range?

Comment (cydrain) [reply internal/util/importutilv2/parquet/field_reader.go:446]: I allocate "n+1" for offsets

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**